Tests/Tracking/Core/UnitTest.swift

```diff
@@ -0,0 +1,76 @@
+@testable import CioDataPipelines
+@testable import CioInternalCommon
+@testable import CioTracking
+import Foundation
+@testable import Segment
+import SharedTests
+
+/// Base class for unit testing within the module, extending `UnitTestBase` with setup and utilities
+/// specific to module components. Ideal for isolated tests of individual functions and classes.
+open class UnitTest: SharedTests.UnitTestBase<CustomerIO> {
+    // Use this `CustomerIO` instance when invoking `CustomerIOInstance` functions in unit tests.
+    // This ensures convenience and consistency across unit tests, and guarantees the correct instance is used for testing.
+    open var customerIO: CustomerIO!
+
+    override open func setUpDependencies() {
+        super.setUpDependencies()
+
+        // Mock date util so the "Date now" is a the same between our tests and the app so comparing Date objects in
+        // test functions is possible.
+        diGraphShared.override(value: dateUtilStub, forType: DateUtil.self)
+        diGraph.override(value: dateUtilStub, forType: DateUtil.self)
+    }
+
+    override open func initializeSDKComponents() -> CustomerIO? {
+        var dataPipelineConfig = DataPipelineConfigOptions.Factory.create(sdkConfig: sdkConfig)
+        // disable auto add destination to prevent tests from sending data to server
+        dataPipelineConfig.autoAddCustomerIODestination = false
+
+        // setup implementation instance for unit tests
+        let implementation = DataPipeline.setUpSharedInstanceForUnitTest(
+            implementation: DataPipelineImplementation(diGraph: diGraphShared, moduleConfig: dataPipelineConfig),
+            config: dataPipelineConfig
+        )
+
+        // setup shared instance with desired implementation for unit tests
+        customerIO = CustomerIO.setUpSharedInstanceForUnitTest(implementation: implementation, diGraph: diGraph)
+        customerIO.setDebugLogsEnabled(sdkConfig.logLevel == .debug)
+
+        return customerIO
+    }
+
+    override open func cleanupTestEnvironment() {
+        super.cleanupTestEnvironment()
+        CustomerIO.resetTestEnvironment()
+    }
+
+    override open func deleteAllPersistantData() {
+        super.deleteAllPersistantData()
+        deleteAllFiles()
+    }
+
+    // function meant to only be in tests as deleting all files from a search path (where app files can be stored!) is
+    // not a good idea.
+    private func deleteAllFiles() {
```

Comment:
Moved this from `SharedTest` to `Tracking` as I think it is only needed by `Tracking` module now